src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -979,22 +975,21 @@ struct ZHighToZLowMatMulOpLowering : public ConversionPattern {
   LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const final {
     Location loc = op->getLoc();
-    ZHighMatMulOp matmulOp = llvm::dyn_cast<ZHighMatMulOp>(op);
     ZHighMatMulOpAdaptor operandAdaptor(operands);
 
     MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
     IndexExprScope scope(create.krnl);
 
-    ZHighMatMulOpShapeHelper shapeHelper(&matmulOp, &rewriter);
-    LogicalResult shapecomputed = shapeHelper.computeShape(operandAdaptor);
-    assert(succeeded(shapecomputed) && "Could not compute output shape");
+    IndexExprBuilderForKrnl createKrnlIE(rewriter, loc);
+    ZHighMatMulOpShapeHelper shapeHelper(op, operands, &createKrnlIE);
```

Comment:
note here: if you create an IndexExprScope outside, you should pass it to the shape helper. Or better, don't create one and use the `shapeHelper.getScope()` to get it. As is, you creating an unused `scope` at the top level which is not used as shapeHelper, when not given any in the constructor, will create its own to save its value. Please check here and elsewhere.

Reply:
Thanks! I removed the unused `scope`.